### PR TITLE
Sasha pasha branch

### PR DESCRIPTION
**Pull request by AlexSyt**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/MapMarkersHelper.java [hunk 1]
```diff
@@ -132,34 +137,37 @@ public int hashCode() {
...
+				R.color.marker_blue,
+				R.color.marker_green,
+				R.color.marker_orange,
+				R.color.marker_red,
+				R.color.marker_yellow,
+				R.color.marker_teal,
+				R.color.marker_purple
+		};
+
+		public static int[] getColors(Context context) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/MapMarkersHelper.java]: This array must be calculated one time only

File: OsmAnd/src/net/osmand/plus/MapMarkersHelper.java [hunk 2]
```diff
@@ -132,34 +137,37 @@ public int hashCode() {
...
-				default:
-					colorId = R.color.marker_blue;
+			return (colorIndex >= 0 && colorIndex < colors.length) ? colors[colorIndex] : colors[0];
+		}
+
+		public static int getColorIndex(Context context, int colorId) {
+			int[] colors = getColors(context);
+			for (int i = 0; i < colors.length; i++) {
+				int color = colors[i];
+				if (color == colorId) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/MapMarkersHelper.java]: Why do you compare color and colorId?

Comment (AlexSyt) [reply OsmAnd/src/net/osmand/plus/MapMarkersHelper.java]: Fixed

File: OsmAnd/src/net/osmand/plus/MapMarkersHelper.java (hunk 1 above)
Comment (AlexSyt) [reply OsmAnd/src/net/osmand/plus/MapMarkersHelper.java]: Fixed

**Pull request merged**